LibreNMS/Permissions.php

```diff
@@ -262,4 +267,21 @@ private function getBillId($bill)
     {
         return $bill instanceof Bill ? $bill->bill_id : (is_numeric($bill) ? (int)$bill : 0);
     }
+
+    /**
+     * @param User|Port $model
+     * @return \Illuminate\Database\Query\Builder
+     */
+    private function getDeviceGroupPermissionsQuery($model)
+    {
+        return DB::table('devices_group_perms')
+        ->select('devices_group_perms.user_id', $model->getQualifiedKeyName())
+        ->join('device_group_device', 'device_group_device.device_group_id', '=', 'devices_group_perms.device_group_id')
+        ->join($model->getTable(), "{$model->getTable()}.device_id", '=', 'device_group_device.device_id')
```

Comment:
This query seems overly complicated.
Why are we joining the ports or devices table?  Seems like the join is only needed for ports.

Reply:
it was without a join before, but since i added support for ports that was the easiest way...

One way would be to simply "inline" the difference in the functions calling this. 
Else we would do something like;
```php
    private function getDeviceGroupPermissionsBaseQuery()
    {
        return DB::table('devices_group_perms')
        ->select('devices_group_perms.user_id')
        ->join('device_group_device', 'device_group_device.device_group_id', '=', 'devices_group_perms.device_group_id')
        ->when(!Config::get('permission.device_group.allow_dynamic'), function ($query) {
            return $query
                ->join('device_groups', 'device_groups.id', '=', 'devices_group_perms.device_group_id')
                ->where('device_groups.type', 'static');
        });
    }

    private function getDeviceGroupDeviceQuery()
    {
        $query = $this->getDeviceGroupPermissionsBaseQuery();

        return $query
            ->select($query->columns[] = 'device_group_device.device_id');
    }

    private function getDeviceGroupDeviceRelatedQuery($model)
    {
        $query = $this->getDeviceGroupPermissionsBaseQuery();

        return $query
            ->select($query->columns[] = $model->getQualifiedKeyName())
            ->join($model->getTable(), "{$model->getTable()}.device_id", '=', 'device_group_device.device_id')

    }    
```